Course/Course/Presentation/Container/CourseContainerViewModel.swift

```diff
@@ -252,15 +252,12 @@ public class CourseContainerViewModel: BaseCourseViewModel {
             isShowProgress = false
             isShowRefresh = false
             
-        } catch let error {
+        } catch {
             isShowProgress = false
             isShowRefresh = false
             shouldShowUpgradeButton = false
-            if error.isInternetError || error is NoCachedDataError {
-                errorMessage = CoreLocalization.Error.slowOrNoInternetConnection
-            } else {
-                errorMessage = CoreLocalization.Error.unknownError
-            }
+            courseStructure = nil
+            courseVideosStructure = nil
```

Comment:
I guess we don't need to set these nil explicitly.

Reply:
These properties are published, and based on these flags, a full-screen error view will appear. Therefore, they are necessary here.